lib/libevent/libevent.nit

```diff
@@ -458,6 +469,35 @@ extern class ConnectionListener `{ struct evconnlistener * `}
 	end
 end
 
+# A listener acting on an Unix Socket, spawns `Connection` on new connections
+extern class UnixConnectionListener `{ struct evconnlistener * `}
+	super ConnectionListener
+
+	private new bind_to(base: NativeEventBase, address: CString, file: CString, factory: ConnectionFactory)
+	import ConnectionFactory.accept_connection, error_callback `{
+		struct sockaddr_un sun;
+		struct evconnlistener *listener;
+
+		unlink(file);
+
+		ConnectionFactory_incr_ref(factory);
+		struct hostent *hostent = gethostbyname(address);
+
+		memset(&sun, 0, sizeof(struct sockaddr_un));
+		sun.sun_family = AF_UNIX;
+		strncpy(sun.sun_path, file, sizeof(sun.sun_path) - 1);
```

Comment:
what is `file` is to big?

Reply:
You mean
- "What if the `char* file` is too big ?"
**OR**
- "What if the path described in `file` is so big that makes it invalid ?"

Reply:
What will happen when the file path is truncated because it couldn't fit in the 108 bytes (or so) available for paths in the `sockaddr_un` structure?
Will it create a random socket somewhere in the middle of the path if it can? How will it be unlinked ?
I'd suggest double checking that `file` is smaller than `sizeof(sun.sun_path)` to avoid such issues before copying the data.

Reply:
UNIX documentation is very specific about the case file is too big. To summarize it says : **DON'T**. Unfortunately, (or fortunately depending on your point of view) such a case would not destroy the program and it will run like it's no big deal, ignoring all errors. I will enforce the 108 chars rule.